utils/post.js

```diff
@@ -309,14 +310,19 @@ function modifyReactionsPost(post, isAnonimous = true) {
   return newPost;
 }
 
-async function filterFeeds(userId, feeds = []) {
+async function filterFeeds(userId, feeds = [], feed_id = null) {
   const newResult = [];
 
   for (const item of feeds || []) {
     const now = moment().valueOf();
     const dateExpired = moment(item?.expired_at).valueOf();
 
-    if (dateExpired < now) continue;
+    if (dateExpired < now) {
+      if (feed_id) {
+        deleteActivityFromUserFeed('topic', feed_id, item.id);
+      }
+      continue;
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for deleting expired feeds from the user feed has been added. This seems fine as long as the `deleteActivityFromUserFeed` function is correctly implemented and handles errors appropriately. However, there's no error handling in case `deleteActivityFromUserFeed` fails. It would be better to wrap it in a try-catch block to handle potential exceptions.

```diff
  if (dateExpired < now) {
    if (feed_id) {
+     try {
        deleteActivityFromUserFeed('topic', feed_id, item.id);
+     } catch (error) {
+       console.error(`Failed to delete activity from user feed: ${error}`);
+     }
    }
    continue;
  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->